azure-iot-device/tests/common/test_async_adapter.py

```diff
@@ -53,27 +80,31 @@ class TestAwaitableCallback(object):
         assert mock_function.call_args == mocker.call()
         assert result == mock_function.return_value
 
+    @pytest.mark.it("Completes the instance Future when a call is invoked on the instance")
     async def test_calling_object_completes_future(self, mock_function):
         callback = async_adapter.AwaitableCallback(mock_function)
         assert not callback.future.done()
         callback()
         await asyncio.sleep(0.1)  # wait to give time to complete the callback
         assert callback.future.done()
 
-    async def test_can_be_called_using_args(self, mocker, mock_function):
+    @pytest.mark.it("Can be called using positional arguments")
+    async def test_can_be_called_using_positional_args(self, mocker, mock_function):
         callback = async_adapter.AwaitableCallback(mock_function)
         result = callback(1, 2, 3)
         assert mock_function.call_count == 1
         assert mock_function.call_args == mocker.call(1, 2, 3)
         assert result == mock_function.return_value
 
-    async def test_can_be_called_using_kwargs(self, mocker, mock_function):
+    @pytest.mark.it("Can be called using explicit keyword arguments")
+    async def test_can_be_called_using_explicit_kwargs(self, mocker, mock_function):
         callback = async_adapter.AwaitableCallback(mock_function)
         result = callback(a=1, b=2, c=3)
         assert mock_function.call_count == 1
         assert mock_function.call_args == mocker.call(a=1, b=2, c=3)
         assert result == mock_function.return_value
 
+    @pytest.mark.it("Can have its callback completion awaited upon")
```

Comment:
>("Can have its callback completion awaited upon") [](start = 19, length = 49)

awkward wording.  I have no idea how to word this better without creating a new test class for the completion method, but I'm not sure if that even makes sense.  maybe "Has a method named completion which returns an awaitable that wraps the associated callback"

Reply:
Like similarly other awkwardly worded stuff, I'm going to re-examine these with a refreshed mind.

---
In reply to: [290982038](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290982038) [](ancestors = 290982038)